### PR TITLE
Fix #2885 Landing page always shows sign up button for private marketplaces

### LINKED ISSUE #2885: Landing page always shows Sign up button for private marketplace, even if the user is logged in
##### Sharetribe Version:

8139d92c778b8dc8c8a8a13fdbbf6634a836a827 (6.1.0)

##### Environment:

Can be reproduced at least in local Mac OSX and sharetribe.com production. 

Node: v6.9.4
Npm: 4.0.1

##### Server mode:

Can be reproduced in development and production modes.

##### Description:

The landing page always shows "Sign up" button in the Hero section for private marketplaces, even if the user is already signed up and logged in.

##### Steps To Reproduce:

1. Create a new marketplace, or use your existing marketplace
2. Make it private (Admin > Settings > check Allow only registered users to see listings and user profiles (make marketplace private)
3. Restart Rails server with Landing page enabled: `clp_static_enabled=true clp_cache_time=0 rails s`
4. Go to the marketplace homepage (landing page)
5. Make sure you are logged in. If not, log in and come back to the landing page.

Expected result: Sign up button should not be shown. Search bar should be shown instead.

Actual result: The sign up button is shown.

screen shot 2017-03-02 at 9 02 56

##### Additional info

Thread on Community forum: https://www.sharetribe.com/community/t/landing-page-always-shows-sign-up/477

### PR DESCRIPTION
**Pull request by rap1ds**
**Problem**: See #2885 for detailed description about the issue

**Solution:**

- Disable browser cache for private marketplaces
- Cache two versions based on the `cta` (which can be either `signup` or `search`) for private marketplace.

### CODE REVIEW

**Comment by rap1ds** [comment]
Uh, tests are failing. I'll fix them

File: app/services/custom_landing_page/caching.rb [hunk 1]
```diff
@@ -3,23 +3,23 @@ module Caching
...
-    def fetch_cache_meta(community_id, version, locale)
-      Rails.cache.read("clp/#{community_id}/#{version}/#{locale}")
+    def fetch_cache_meta(community_id, version, locale, cta)
+      Rails.cache.read("clp/#{community_id}/#{version}/#{locale}/#{cta}")
     end
 
-    def fetch_cached_content(community_id, version, digest)
-      Rails.cache.read("clp/#{community_id}/#{version}/#{digest}")
+    def fetch_cached_content(community_id, version, digest, cta)
+      Rails.cache.read("clp/#{community_id}/#{version}/#{digest}/#{cta}")
```
Comment (bladealslayer) [inline app/services/custom_landing_page/caching.rb]: I'm not sure you need `cta` in the content cache keys. The content is different, so the digest will be different. As long as you have the meta based on the correct `cta`, it should be fine.

File: app/services/custom_landing_page/caching.rb [hunk 2]
```diff
@@ -3,23 +3,23 @@ module Caching
...
     end
 
-    def cache_content!(community_id, version, locale, content, cache_time)
+    def cache_content!(community_id, version, locale, content, cache_time, cta)
```
Comment (bladealslayer) [inline app/services/custom_landing_page/caching.rb]: Same as above

File: app/services/custom_landing_page/caching.rb [hunk 3]
```diff
@@ -30,13 +30,13 @@ def build_cache_meta(content)
...
-      Rails.cache.write("clp/#{community_id}/#{version}/#{locale}", cache_meta, expires_in: cache_time)
+    def write_cache_meta!(community_id, version, locale, cache_meta, cache_time, cta)
+      Rails.cache.write("clp/#{community_id}/#{version}/#{locale}/#{cta}", cache_meta, expires_in: cache_time)
     end
 
     ## Internal, use cache_content! instead
-    def write_cached_content!(community_id, version, content, digest, cache_time)
-      Rails.cache.write("clp/#{community_id}/#{version}/#{digest}", content, expires_in: cache_time)
+    def write_cached_content!(community_id, version, content, digest, cache_time, cta)
+      Rails.cache.write("clp/#{community_id}/#{version}/#{digest}/#{cta}", content, expires_in: cache_time)
```
Comment (bladealslayer) [inline app/services/custom_landing_page/caching.rb]: Same as above.

File: app/services/custom_landing_page/caching.rb [hunk 4]
```diff
@@ -30,13 +30,13 @@ def build_cache_meta(content)
...
-    def write_cache_meta!(community_id, version, locale, cache_meta, cache_time)
-      Rails.cache.write("clp/#{community_id}/#{version}/#{locale}", cache_meta, expires_in: cache_time)
+    def write_cache_meta!(community_id, version, locale, cache_meta, cache_time, cta)
+      Rails.cache.write("clp/#{community_id}/#{version}/#{locale}/#{cta}", cache_meta, expires_in: cache_time)
     end
 
     ## Internal, use cache_content! instead
     def write_cached_content!(community_id, version, content, digest, cache_time)
-      Rails.cache.write("clp/#{community_id}/#{version}/#{digest}", content, expires_in: cache_time)
+      Rails.cache.write("clp/#{community_id}/#{version}/#{digest}/", content, expires_in: cache_time)
```
Comment (bladealslayer) [inline app/services/custom_landing_page/caching.rb]: Runaway trailing `/`

**Pull request merged**